cloudera/datadog_checks/cloudera/api_client_v7.py

```diff
@@ -19,17 +20,38 @@ def collect_data(self):
         read_clusters_response = clusters_resource_api.read_clusters(cluster_type='any', view='full')
         self._log.debug("Cloudera full clusters response:\n%s", read_clusters_response)
 
+        if self._check.config.clusters:
+            self._log.debug("clusters config: %s", self._check.config.clusters)
+            d = Discovery(
+                lambda: clusters_resource_api.read_clusters(cluster_type='any', view='full').items,
+                limit=self._check.config.clusters.limit,
+                include=self._check.config.clusters.include,
+                exclude=self._check.config.clusters.exclude,
+                interval=self._check.config.clusters.interval,
+                key=lambda cluster: cluster.name,
+            )
+            discovered_clusters = list(d.get_items())
+        else:
+            discovered_clusters = [
+                (None, cluster.name, cluster, None)
+                for cluster in clusters_resource_api.read_clusters(cluster_type='any', view='full').items
+            ]
```

Comment:
What do you think of also using your new `Discovery` utility in this case but without specifying any include/exclude/limit to leverage the caching and refresh features it has?

Reply:
I am not sure of understanding what do you mean...

Reply:
Sorry if that was not clear. What I mean is that in this case, you are calling `clusters_resource_api.read_clusters(cluster_type='any', view='full').items` directly, which means that once the integration in set up, the cluster list stored in `discovered_clusters` won't change. Because of that, if a new cluster is added to the cloudera instance, the customer will have to restart their agent to take this new cluster into account. 

Instead of that, you could use the Discovery class you just added to have a way to "refresh" this cluster list from time to time, without needing to restart the agent. You could do something like that: 

```python
            d = Discovery(
                lambda: clusters_resource_api.read_clusters(cluster_type='any', view='full').items,
                limit=None, # No limit, get them all
                include=None,
                exclude=None, 
                interval=60, # arbirtrary defined for the example
                key=lambda cluster: cluster.name,
            )
```

This way the cluster list will be refresh from time to time (here every 60 secs) but we are still going to get all of them

Reply:
`clusters_resource_api.read_clusters(cluster_type='any', view='full').items` is called each time the check runs, so if a new cluster is added to the `cloudera` instance we will get it...

Reply:
Ok thanks for the answer, so I guess my assumption [here](https://github.com/DataDog/integrations-core/pull/13664/files#r1065795030) is correct